ewfish.py

```diff
@@ -0,0 +1,980 @@
+			response = "You're too hungry to fish right now."
+
+		else:
+			has_fishingrod = False
+
+			if user_data.weapon >= 0:
+				weapon_item = EwItem(id_item = user_data.weapon)
+				weapon = ewcfg.weapon_map.get(weapon_item.item_props.get("weapon_type"))
+				if weapon.id_weapon == "fishingrod":
+					has_fishingrod = True
+
+			fisher.current_fish = gen_fish(market_data, cmd, has_fishingrod)
+			fisher.fishing = True
+			fisher.bait = False
+			fisher.pier = user_data.poi
+			item_search = ewutils.flattenTokenListToString(cmd.tokens[1:])
+			author = cmd.message.author
+			server = cmd.message.server
+
+			item_sought = ewitem.find_item(item_search = item_search, id_user = author.id, id_server = server.id)
+
+			if item_sought:
+				item = EwItem(id_item = item_sought.get('id_item'))
+				str_name = item = EwItem(id_item = item_sought.get('str_name'))
+
+				if item.item_type == ewcfg.it_food:
+					fisher.bait = True
+
+					if item in ewcfg.plebe_bait:
+						fisher.current_fish = "plebefish"
+
+					elif item == "doublestuffedcrust":
+						if random.randrange(5) == 3:
+							fisher.current_fish = "doublestuffedflounder"
+
+					elif item in ["chickenbucket", "familymeal"]:
+						if random.randrange(5) == 3:
+							fisher.current_fish = "seacolonel"
+
+					elif item in ["steakvolcanoquesomachorito", "nachosupreme"]:
+						if random.randrange(5) == 3:
+							fisher.current_fish = "marlinsupreme"
+
+					elif item in ["blacklimes", "blacklimesour"]:
+						if random.randrange(1) == 1:
+							fisher.current_fish = "blacklimesalmon"
+
+					elif item in ["pinkrowddishes", "pinkrowdatouille"]:
+						if random.randrange(1) == 1:
+							fisher.current_fish = "thrash"
+
+					elif item in ["purplekilliflowercrustpizza", "purplekilliflower"]:
+						if random.randrange(1) == 1:
+							fisher.current_fish = "dab"
+
+					elif item == "kingpincrab":
+						if random.randrange(5) == 1:
+							fisher.current_fish = "kingpincrab"
+
+					elif float(item.time_expir if item.time_expir is not None else 0) < time.time():
+						if random.randrange(1) == 1:
+							fisher.current_fish = "plebefish"
+					ewitem.item_delete(item_sought.get('id_item'))
+
+			if fisher.current_fish == "item":
+				fisher.current_size = "item"
+
+			else:
+				fisher.current_size = gen_fish_size(has_fishingrod)
+
+			if fisher.bait == False:
+				response = "You cast your fishing line into the "
+			else:
+				response = "You attach your {} to the hook as bait and then cast your fishing line into the ".format(str_name)
+
+
+			if cmd.message.channel.name in [ewcfg.channel_afb_pier, ewcfg.channel_vc_pier, ewcfg.channel_se_pier, ewcfg.channel_ferry]:
+				response += "vast Slime Sea."
+			else:
+				response += "glowing Slime Lake."
+
+			await ewutils.send_message(cmd.client, cmd.message.channel, ewutils.formatMessage(cmd.message.author, response))
+
+			user_data.hunger += ewcfg.hunger_perfish * ewutils.hunger_cost_mod(user_data.slimelevel)
+			user_data.persist()
```

Comment:
never persist an object after an await call. other code may run during that time, which may make changes to this user, which we would then overwrite